File: backend/src/main/java/springweb/backend/model/Item.java
package springweb.backend.model;

import lombok.Builder;
import lombok.With;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Document("items")
@With
@Builder
public record Item(
        @Id String id,
        String name,
        String img,
        String description,
        ItemCategory category,
        ItemStatus status,
        List<Double> geocode,
        String owner
) {
}

File: frontend/src/models/Item.ts
import {ItemCategory} from "./ItemCategory.ts";
import {ItemStatus} from "./ItemStatus.ts";

export type Item = {
    id: string,
    name: string,
    img: string,
    description: string,
    category: ItemCategory,
    status: ItemStatus,
    geocode: [number,number],
    owner: string
}